app/src/main/kotlin/compiler/Compiler.kt

```diff
@@ -36,6 +36,8 @@ class Compiler(val diagnostics: Diagnostics) {
             val programProperties = Resolver.resolveProgram(ast, diagnostics)
 
             // TODO: generate the code
-        } catch (_: CompilationFailed) { }
+        } catch (_: CompilationFailed) { } finally {
+            println(diagnostics)
```

Comment:
This doesn't have to be in a `finally` block - if there is an internal error, then we are in trouble anyway.